report/node.go

```diff
@@ -182,8 +184,26 @@ func (n Node) WithChildren(children NodeSet) Node {
 
 // WithChild returns a fresh copy of n, with one child merged in.
 func (n Node) WithChild(child Node) Node {
-	n.Children = n.Children.Merge(MakeNodeSet(child))
-	return n
+	result := n.Copy()
+	result.Children = result.Children.Merge(MakeNodeSet(child))
+	return result
+}
+
+// Copy returns a value copy of the Node.
+func (n Node) Copy() Node {
```

Comment:
Yes that commit is different from mine, but they have the same title and description. Something went wrong.